p2p/transport.go

```diff
@@ -595,7 +595,10 @@ func resolveIPs(resolver IPResolver, c net.Conn) ([]net.IP, error) {
 		return nil, err
 	}
 
-	addrs, err := resolver.LookupIPAddr(context.Background(), host)
+	timeoutCtx, cancelFunc := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
Is this value, a one-second timeout for name resolution reasonable? I feel that DNS server queries may sometimes take longer, and considering that UDP packets may be missed, shouldn't retry also be considered?

Reply:
I found following reference.
https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/dns-client-resolution-timeouts

Time (seconds since start) | Action
-- | --
0 | Client queries the DNS server
1 | If no response is received after 1 second, client queries again the DNS server
2 | If no response is received after 1 more second, client queries again the DNS server
4 | If no response is received after 2 more seconds, client queries again the DNS server
8 | If no response is received after 4 more seconds, client queries again the DNS server
10 | If no response is received after 2 more seconds, client stops querying

What about this rule? I think it could be simplified a bit.

Reply:
Simple rules: timeout is 2 seconds, retry up to 4 times (5 queries in total)
What about this?